src/ring_theory/polynomial/opposites.lean

```diff
@@ -0,0 +1,90 @@
+/-
+Copyright (c) 2022 Damiano Testa. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Damiano Testa
+-/
+
+import data.polynomial.induction
+import data.polynomial.degree.definitions
+
+/-!  #  Interactions between `R[X]` and `Rᵐᵒᵖ[X]`
+
+This file contains the basic API for "pushing through" the isomorphism
+`op_ring_equiv : R[X]ᵐᵒᵖ ≃+* Rᵐᵒᵖ[X]`.  It allows going back and forth between a polynomial ring
+over a semiring and the polynomial ring over the opposite semiring. -/
+
+open_locale polynomial
+open polynomial mul_opposite
+
+variables {R : Type*} [semiring R] {p q : R[X]}
+
+noncomputable theory
+
+namespace polynomial
+
+/-- Ring isomorphism between `R[X]ᵐᵒᵖ` and `Rᵐᵒᵖ[X]` sending each coefficient of a polynomial
+to the corresponding element of the opposite ring. -/
+@[simps]
```

Comment:
I think probably you don't want this `simps` at all, the lemmas it generates are likely worse than the ones you create below.

Reply:
I removed the `simps` tag, although I do not know what it does.